datastream-kafka-connector/src/main/java/com/linkedin/datastream/connectors/kafka/AbstractKafkaBasedConnectorTask.java

```diff
@@ -388,7 +388,7 @@ public void run() {
       consumerSubscribe();
 
       ConsumerRecords<?, ?> records;
-      while (!_shutdown) {
+      while (!_shutdown && !Thread.currentThread().isInterrupted()) {
```

Comment:
The code path trying to interrupt the thread, does it not call shutdown() method which sets _shutdown to true?

Reply:
Can you pls explain the code flow for how you think the interrupted exception is being lost (or more likely being caught and then ignored)?